_posts/2024-10-15-anaconda-defaults.md

```diff
@@ -0,0 +1,139 @@
+
+**tldr:** Run these commands:
+
+```
+conda config --show channels
+conda config --remove channels defaults
+conda config --show channels
+```
+
+## Background
+
+Ananconda Inc has started to threaten [legal action against against commercial companies](https://www.reuters.com/legal/litigation/intel-sued-copyright-infringement-over-ai-software-2024-08-09/) and [has also advised non-profits](https://www.theregister.com/2024/08/08/anaconda_puts_the_squeeze_on/) to purchase licenses to the anaconda software/distribution channels. While the situation is being resolved, some institutions have blocked the `anaconda.org` domain completely.
+
+This has been brewing for a number of years. The first change happened [back in 2020](https://www.anaconda.com/blog/sustaining-our-stewardship-of-the-open-source-data-science-community), and the second happened very [recently in March 2024](https://legal.anaconda.com/policies/en/?name=terms-of-service#anaconda-terms-of-service) that affects non profit organisations ["government entities and non-profit entities with over 200 employees or contractors"](https://www.theregister.com/2024/08/08/anaconda_puts_the_squeeze_on/). This is problematic due the wording around 'employees' - many organisations have hundreds of employees but few users of `conda` software.
+
+Before we dive in, a quick recap definitions:
+
+- Anaconda Inc is the commercial entity behind the `conda` and `miniconda` software and the `Anaconda.Navigator` software suite.
+- Anaconda has curated packages which are available [through specific channels](https://repo.anaconda.com/pkgs/)  
+- `Miniforge` is a non-Anaconda installer specific to `conda-forge` channel, 
+- `mamba` is a drop in replacement to `conda`.
+- Far more detailed information on all the different channels/distributions [can be found here.](https://bioconda.github.io/faqs.html#what-s-the-difference-between-anaconda-conda-miniconda-mamba-mambaforge-micromamba)
+
+All the hooha surrounds the [curated Ananconda channels](https://docs.anaconda.com/working-with-conda/reference/default-repositories/), the primary one being the `defaults` channel. **When installing `conda/miniconda` software, the `defaults` channel is added to your global channels list.** You cond inadvertantly be using Anaconda services without intending to.
+
+## How to's
+
+Below is guidance on how to best deal with `defaults` channel.
+
+### Fresh Install - Best practices
+
+1. Install `miniforge` conda-forge specific distribution  - [instructions here](https://github.com/conda-forge/miniforge), this will also install mamba.
+2. Add the channels `bioconda` and `nodefaults` [in that order] as global defaults
+```
+conda config --add channels bioconda nodefaults
+```
+That's it!
+
+### Current install - Remove `defaults`:
+
+1. To check if your installation has `defaults` channel in your global configuration:
+
+```
+conda config --show channels
+# or more informative: 
+conda config --get
+```
+
+2. Now remove `defaults`:
```

Comment:
```suggestion
2. Now remove `defaults`:
```

You might want to indent these code blocks so they stay inline with the bullets of the list

Reply:
Does this affect the installation itself or just the users `.condarc? file?

Reply:
It removes defaults from the `.condarc`

Reply:
@pmenzel that's a good point, should clarify that it doesnt affect the currently installed packages and does not affect the hard coded defaults.